modules/ardrivo/src/Servo.cxx

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Servo.cxx
+ *  Copyright 2020 ItJustWorksTM
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include <Arduino.h>
+#include "BoardDataDef.hxx"
+#include "Servo.h"
+
+#define SERVO_MIN() (MIN_PULSE_WIDTH - this->min * 4)
+#define SERVO_MAX() (MAX_PULSE_WIDTH - this->max * 4)
+
+#define TRIM_DURATION 2
+
+uint32_t servoCount = 0;
```

Comment:
Do not use a mutable global

Reply:
Not sure what you mean

Reply:
Globals should all be `constexpr` qualified (except `board_data` and `board_info`)

Reply:
Also, why in the world do you count the number of servos? we have no limit as to how many servos we support

Reply:
I don't know. It's there in the official library. That's why I included it here. I think that's for counting the index of servo in the vector.

Reply:
If you don't know why it's there, then why are you even using it?

Reply:
Cause how are you gonna keep track of the index of the last servo you adding in?

Reply:
We don't even have a list of servos
`BoardData::servo_value` has one entry per pin of the board, and holds the angle value that a servo attached to said pin would read-from/write-to it

Reply:
You mean, for example, the angle value of index 5 will be `servo_value[4] `. I think I got the right idea for that. what do you mean by one entry per pin?

Reply:
No; `servo_value[12]` contains the angle that a servo would read/write if it is attached to the pin n°12

Reply:
We don't even need the constructor then.

Reply:
I believe we do; I'm curious what kind of rabbit you'll take out of your hat to properly initialize objects of type `Servo`

Reply:
We could assign the default value for the angle of servo. But since there is no pin attach on it. What kind of logic we should put in it?